src/app/modules/main/module.nim

```diff
@@ -1256,3 +1258,31 @@ method windowActivated*[T](self: Module[T]) =
 
 method windowDeactivated*[T](self: Module[T]) =
   self.controller.speedupArchivesImport()
+
+method parseCommunitySharedUrl*[T](self: Module[T], url: string): string =
+  let communityData = self.controller.parseCommunitySharedUrl(url)
+  var jsonObj = newJObject()
+  jsonObj["displayName"] = %* communityData.displayName
+  jsonObj["description"] = %* communityData.description
+  jsonObj["membersCount"] = %* communityData.membersCount
+  jsonObj["color"] = %* communityData.color
+  jsonObj["communityId"] = %* communityData.communityId
+  return $jsonObj
+
+method parseCommunityChannelSharedUrl*[T](self: Module[T], url: string): string =
+  let channelData = self.controller.parseCommunityChannelSharedUrl(url)
+  var jsonObj = newJObject()
+  jsonObj["displayName"] = %* channelData.displayName
+  jsonObj["description"] = %* channelData.description
+  jsonObj["emoji"] = %* channelData.emoji
+  jsonObj["color"] = %* channelData.color
+  jsonObj["uuid"] = %* channelData.uuid
+  return $jsonObj
+
+method parseContactSharedUrl*[T](self: Module[T], url: string): string =
+  let contactData = self.controller.parseContactSharedUrl(url)
+  var jsonObj = newJObject()
+  jsonObj["displayName"] = %* contactData.displayName
+  jsonObj["description"] = %* contactData.description
+  jsonObj["publicKey"] = %* contactData.publicKey
+  return $jsonObj
```

Comment:
Can we put those in the `urls_manager` instead?
I think we just need to expose it to the QML using `setRootContextProperty`.
Though I think it can't have access to services.

Anyway, maybe someone has an idea where we could put them? Putting them in the main module works, it just feels like it's a bit random

Reply:
Having a url service and mvc makes sense

Reply:
so probably, I should move generators to new url service as well? otherwise, its service with just one method. @jrainville @MishkaRogachev what do you think?

Reply:
Yeah, I do believe that urls could become massive in the future